Edit PR comment in place instead of creating a new comment

palexcast opened the pull request:
Added prefix `<!-- im-open/process-dotnet-test-results -->` to comments.
We can then use this to check if any comment starts with the comment, if it does we rather update this comment then create a new comment.
If no comment exists with this prefix we create a new comment.

danielle-casella-adams reviewed (changes requested):
This is significantly changing the way this action behaves.  I think for these changes to be incorporated, it would need a new flag so the user can indicate whether they always want a new comment or to update an existing one if one is found.

palexcast commented:
@danielle-casella-adams I agree fully, I'll update the PR and notify you.

palexcast commented:
@danielle-casella-adams I've updated the PR with a new flag and documentation for it.

danielle-casella-adams commented on README.md:
```diff
@@ -42,14 +42,15 @@ For failed test runs you can expand each failed test and view more details about
+| `github-token`                | true        | N/A                              | Used for the GitHub Checks API.  Value is generally: secrets.GITHUB_TOKEN.                                              |
+| `base-directory`              | false       | `.` Root Directory of repository | The base directory of where to look for `trx` files.                                                                    |
+| `create-status-check`         | false       | true                             | Flag indicating whether a status check with code coverage results should be generated.                                  |
+| `create-pr-comment`           | false       | true                             | Flag indicating whether a PR comment with code coverage results should be generated.                                    |
```
```suggestion
| `create-pr-comment`           | false       | true                             | Flag indicating whether a PR comment with code coverage results should be generated.  When `true` the default behavior is to always create a new comment.|
```

danielle-casella-adams commented on README.md:
```diff
@@ -42,14 +42,15 @@ For failed test runs you can expand each failed test and view more details about
+| `base-directory`              | false       | `.` Root Directory of repository | The base directory of where to look for `trx` files.                                                                    |
+| `create-status-check`         | false       | true                             | Flag indicating whether a status check with code coverage results should be generated.                                  |
+| `create-pr-comment`           | false       | true                             | Flag indicating whether a PR comment with code coverage results should be generated.                                    |
+| `update-pr-comment-on-change` | false       | false                            | Flag indicating whether the PR comment should be updated upon changes.                                                  |
```
```suggestion
| `update-comment-if-one-exists` | false       | false                            | When `update-pr-comment-on-change` is true, this flag determines whether a new comment is created (default) or if the action updates an existing comment if one is found.|
```

palexcast replied:
Updated documentation and variable name

danielle-casella-adams commented on src/main.js:
```diff
@@ -8,6 +8,7 @@ const baseDir = core.getInput('base-directory') || '.';
 const ignoreTestFailures = core.getInput('ignore-test-failures') == 'true';
 const shouldCreateStatusCheck = core.getInput('create-status-check') == 'true';
 const shouldCreatePRComment = core.getInput('create-pr-comment') == 'true';
+const shouldUpdateCommentOnChange = core.getInput('update-pr-comment-on-change') == 'true';
```
I do have a strong preference for the naming of the variable, sorry I was out Friday and didn't see your question until today.

```suggestion
const updateCommentIfOneExists = core.getInput('update-comment-if-one-exists') == 'true';
```

palexcast replied:
👍

danielle-casella-adams commented on src/github.js:
```diff
@@ -36,26 +36,69 @@ async function createStatusCheck(repoToken, reportData, markupData, conclusion)
+    return;
+  }
+
+  const prefixedMarkupData = '<!-- im-open/process-dotnet-test-results -->' + markupData;
```
You probably need to add a newline here between the comment and the markupData, otherwise the first item in the markup data (a heading) won't render properly.

palexcast replied:
I've added a newline

danielle-casella-adams commented:
@palexcast I do have a strong preference for the naming.  I want it to be very clear/explicit what the variable is doing and when it should be used.  I made a couple of suggestions for the readme inputs section and I think one comment in the js.  Obviously all references to that variable will need to be updated and a new npm run build/format needs to be run.

palexcast commented:
@danielle-casella-adams I have updated the comment per your suggestions.
I agree that the new variable name was a bit more "direct". 
Added the new line and updated the documentation were necessary after renaming the variable.

danielle-casella-adams commented on README.md:
```diff
@@ -42,15 +42,15 @@ For failed test runs you can expand each failed test and view more details about
+| `base-directory`               | false       | `.` Root Directory of repository | The base directory of where to look for `trx` files.                                                                                                                       |
+| `create-status-check`          | false       | true                             | Flag indicating whether a status check with code coverage results should be generated.                                                                                     |
+| `create-pr-comment`            | false       | true                             | Flag indicating whether a PR comment with code coverage results should be generated. When `true` the default behavior is to always create a new comment.                   |
+| `update-comment-if-one-exists` | false       | false                            | When `update-comment-if-one-exists` is true, this flag determines whether a new comment is created (default) or if the action updates an existing comment if one is found. |
```
Sorry, I gave you a bad suggestion the first time in the review.
```suggestion
| `update-comment-if-one-exists` | false       | false                            | When `create-pr-comment` is true, this flag determines whether a new comment is created (default) or if the action updates an existing comment if one is found. |
```

danielle-casella-adams commented on README.md:
```diff
@@ -42,15 +42,15 @@ For failed test runs you can expand each failed test and view more details about
 <kbd><img src="./docs/failed_tests.png"></img></kbd>
 
```
I can't make a comment on the correct line, but in the readme example you'll need to update the version of this action to v2.0.0.   When you commit that change please add `+semver:major` to the commit message so the increment-version-on-merge workflow knows how to increment this appropriately.

danielle-casella-adams reviewed (changes requested):
Sorry one more request and the other change was my fault, I gave you a bad suggestion in the first PR.  I appreciate you making these changes though!

danielle-casella-adams commented:
I also just noticed the action.yml file was not updated with the new argument.

danielle-casella-adams commented:
You can disregard the last couple of comments.  I'm going to introduce the same kinds of changes into the other actions that work just like this one and can make a few small tweaks to these changes at the same time.  Thank you for this PR!

palexcast commented:
Oh! Sorry I should have at least remembered to update the action.yml!
No problem, and I appreciate the feedback :)!
Looking forwards to continue using this action 👍 !
Thank you for creating it!

danielle-casella-adams commented:
@palexcast We have a number of other actions that do similar things for jest/postman/cypress/code coverage summaries, so I took what you've done here and updated it slightly.  I used the same idea in our other actions as well.  If you want to see the changes, you can view them here: #7.  

Thank you again for your work on this, it was very timely and I had my own TODO to get something similar implemented.
